### PR TITLE
feat: add destination status and last seen to CLI and UI

### LINKED ISSUE #2151: Need to show destination status to know if a cluster is still alive
Currently there's no way for users to know if a cluster is "active" or not – i.e. does it still exist and is it still connected to Infra. We used to use the `updated` field for this, but it should really be its own field since the cluster may not change state.

Example request:

```
GET /api/destinations/<id>

{
  ...
  "last_seen": "2022-05-31T14:43:56+00:00"
  ...
}
```

- [ ] Add `Infra-Connector-Unique-ID` header to requests from the connector to the server
- [ ] Add and update `LastSeenAt` Destination field when an authenticated request with the header is received by the server

### PR DESCRIPTION
**Pull request by mxyng**
## Summary

- Add Destination `Status` API field
- Add columns for Status and Last Seen to CLI and UI list outputs

```
$ infra destinations list
  NAME            URL                 STATUS        LAST SEEN
  docker-desktop  localhost:123       Disconnected  never
  docker-desktop  localhost:123       Disconnected  never
  docker-desktop  10.98.125.93:10443  Connected     1 second ago
```
image

## Checklist

- [ ] Wrote appropriate unit tests
- [ ] Considered security implications of the change
- [ ] Updated associated docs where necessary
- [ ] Updated associated configuration where necessary
- [ ] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [ ] Nothing sensitive logged
- [ ] Considered data migrations for smooth upgrades

## Related Issues

Resolves #2151

### CODE REVIEW

File: ui/pages/destinations/index.js [hunk 1]
```diff
@@ -290,13 +290,33 @@ const columns = [
...
-    width: '33%',
+    width: '25%',
     Cell: ({ value }) => (
       <span className='rounded bg-gray-800 px-2 py-0.5 text-gray-400'>
         {value.kind}
       </span>
     ),
   },
+  {
+    Header: 'Status',
```
Comment (mxyng) [inline ui/pages/destinations/index.js:301]: This doesn't necessarily need a column. An alternative may be to add it as an icon in the `Last Seen` column @FSHA

File: ui/pages/destinations/index.js [hunk 2]
```diff
@@ -305,13 +305,27 @@ const columns = [
...
+    accessor: v => v,
+    width: '15%',
+    Cell: ({ row, value }) => (
+      <div className='items-center py-2'>
```
Comment (jmorganca) [inline ui/pages/destinations/index.js:320]: Missing `flex`

File: internal/server/models/destination.go [hunk 3]
```diff
@@ -21,6 +21,12 @@ type Destination struct {
 }
 
 func (d *Destination) ToAPI() *api.Destination {
+	connected := false
+	// TODO: this should be configurable
```
Comment (jmorganca) [inline internal/server/models/destination.go:25]: Nit: create an issue for this TODO?

File: internal/cmd/destinations.go [hunk 4]
```diff
@@ -58,15 +58,24 @@ func newDestinationsListCmd(cli *CLI) *cobra.Command {
...
-					URL  string `header:"URL"`
+					Name     string `header:"NAME"`
+					URL      string `header:"URL"`
+					Status   string `header:"STATUS"`
+					LastSeen string `header:"LAST SEEN"`
 				}
 
 				var rows []row
 				for _, d := range destinations.Items {
+					status := "Disconnected"
```
Comment (jmorganca) [inline internal/cmd/destinations.go:69]: Nit: make these constants / enums?

**Review by jmorganca — APPROVED** [review]
LGTM – two smaller comments. Nice work!

**Pull request merged**